options/posix/generic/unistd-stubs.cpp

```diff
@@ -15,9 +15,10 @@
 #include <mlibc/posix-sysdeps.hpp>
 #include <mlibc/thread.hpp>
 
-unsigned int alarm(unsigned int) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+unsigned int alarm(unsigned int seconds) {
+	struct itimerval it = {it.it_value.tv_sec = seconds}, old = {0};
+	setitimer(ITIMER_REAL, &it, &old);
+	return old.it_value.tv_sec + !! old.it_value.tv_usec;
 }
```

Comment:
What if `setitimer` fails here?

Reply:
Musl doesn’t seem to handle that explicitly, and I guess setitimer takes care of errno in that case. On that note, the man page of alarm doesn’t list what should happen with respect to errors, nor does it list valid errno’s.

Reply:
Hmm I’m not so sure about that. We could check the return value of setitimer and return 0 if it failed, but flat out crashing seems not the right solution.

Reply:
There are only two failing conditions for ``setitimer()``, which is if the pointers aren't valid or ``which`` is not one of ``ITIMER_REAL``, ``ITIMER_VIRTUAL``, or ``ITIMER_PROF``, neither of which happens in this case.

Reply:
There are only two failing conditions for ``setitimer()``, which is if the pointers aren't valid or ``which`` is not one of ``ITIMER_REAL``, ``ITIMER_VIRTUAL``, or ``ITIMER_PROF``, neither of which happens in this case.